### ISSUE TITLE
Maintain Sqlite config database in mounted storage container

### ISSUE THREAD

**Issue by thekaveman** [cal-itp/benefits#1673]
This is the follow-up to [#1312](https://github.com/cal-itp/benefits/issues/1312#issuecomment-1615008764), where we looked at some different options to enable longer-term storage of our configuration data, a pre-cursor to the Admin interface feature. 

Rather than get weighed down by the extra complexity of a new database service, we decided:

> > Some kind of syncing to maintain the Sqlite database between container restarts
> 
> This is probably the simplest approach with the fewest number of changes, and could be done with no additional services in Azure (cost may increase slightly for additional blob storage).

## Acceptance Criteria

- [ ] `bin/init.sh` no longer deletes the database (maybe hide this behind a flag?)
  - [ ] Put the call to create the `superuser` behind the same flag
- [ ] `django.db` stored in a location that is mounted to a (new?) storage container in the Azure environment
- [ ] Storage container has some backup/snapshotting configured to allow at least a daily backup/rollback of the database
- [ ] Local devcontainer workflow is unaffected / still works OOTB

## Additional context / questions

- [ ] Can we change the path where the `django.db` database is (initially) created, to the mount? Yes, this should be fine in `benefits/settings.py`
- [ ] As of now we've only mounted for _reading_ files; is it possible to _write_ to files in the mount?
- [ ] How do we handle the existing data migrations?
  - Remove it and seed the database another way when it is re-created?
  - Maintain it and build upon with future migrations? Probably the safest idea to keep everything working.

**Comment by machikoyasuda** [comment]
Started researching for this ticket. These are really old StackOverflows, so I'm not sure if it's still current - but there seems to be a consensus that you can't store a SQLite database on an Azure Blob Storage or File Storage, you need a Azure Web App (https://azure.microsoft.com/en-us/products/app-service/web#features) or serverless.

https://stackoverflow.com/questions/37317754/where-to-put-sqlite-database-file-in-azure-app-service
https://stackoverflow.com/questions/43546946/hosting-sqlite-db-on-azure-storage
https://stackoverflow.com/questions/76166628/can-i-store-and-then-read-write-to-a-sqlite-database-using-azure-file-storage-fr

@thekaveman Just wanted to confirm that's okay with the direction we're going in.

**Comment by thekaveman** [comment]
@machikoyasuda I'd like a little more research here before we say this isn't the right approach or is not possible. Some additional thoughts:

* We do run an Azure Web App, it is a Linux Web App, you can see this in the [terraform definition](https://github.com/cal-itp/benefits/blob/dev/terraform/app_service.tf#L13)
* We know that mounting a file from a storage container into the running app, where the app can read that file, works -- e.g. this is how we get the Courtesy Card data into Eligibility Server
* Sqlite does now natively support cloud-backed storage accounts, more info here (very technical with C code, we won't be writing that but maybe a version is out there that is already setup like this): https://sqlite.org/cloudsqlite/doc/trunk/www/index.wiki
* This article series (2) talks about mounting an _Azure File Share_ to read Sqlite data from an app, slightly different from an _Azure Storage Container_ -- maybe this approach would work instead? https://zimmergren.net/mount-an-azure-storage-file-share-to-deployments-in-azure-kubernetes-services-aks/
* Maybe there's some other way to do this that doesn't depend on an active connection to a share/storage container? E.g. if we can get and act on a signal of when the app service is shutting down, and sync the Sqlite file from the app service filesystem back into the storage container at that time? Then read it back out into the app service filesystem the next time it starts up? So the app isn't trying to read/write live from the storage container.

I want us to consider the "full database service in Azure" as our last resort and fully exhaust the Sqlite approach before we go that route. Maybe timebox to the rest of the Sprint (so next week) for the research, and if nothing comes of it then we switch tactics.

Definitely favor newer articles/SO posts, things have been changing a lot in this space. I'd say any advice before 2020 should probably be de-prioritized (while acknowledging the Azure File Share article I posted above is from 2019).

**Comment by angela-tran** [comment]
I'm wondering if there's a low-cost, low-risk way for us to test out these different ideas for maintaining our SQLite database across app restarts. We've read through a lot of information that may or may not be entirely up-to-date / correct, so I think the best way to know what actually works is to try it out.

**Comment by angela-tran** [comment]
Would it be ok to set up a new resource group with the minimal amount of resources to test out our ideas? Or if not, should we use the dev resource group?

**Comment by machikoyasuda** [comment]
I agree with @angela-tran's idea of setting up a new resource group with the least amount of resources to test out costs/configuration options.

**Comment by thekaveman** [comment]
@machikoyasuda just noticed this line from the most recent SO post in your list: https://stackoverflow.com/a/76167664/453168

> Therefore, given your scenario, to keep it simple and cheap, my suggestion to avoid complexity by using other Azure Services to persist your database is to save your SQLite file inside /home/site/wwwroot folder as mentioned [here](https://stackoverflow.com/questions/37317754/where-to-put-sqlite-database-file-in-azure-app-service).

We should look at this as a first step, just moving where the Sqlite file is stored (from `/tmp` to `/home/site/wwwroot`).

Later we can look at backing up and restoring this file to more permanent storage.
